### ISSUE TITLE
compile errors

### ISSUE THREAD

**Issue by jyasuu** [jyasuu/llm-can-schedule#4]
Compiling jssp-scheduler-real v0.1.0 (/mnt/d/git/llm-can-schedule/integrated-real-train)
warning: use of deprecated function `rand::thread_rng`: Renamed to `rng`
   --> src/main.rs:321:26
    |
321 |     let mut _rng = rand::thread_rng();
    |                          ^^^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: use of deprecated function `rand::thread_rng`: Renamed to `rng`
  --> src/scheduler.rs:30:29
   |
30 |         let mut rng = rand::thread_rng();
   |                             ^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> src/model.rs:29:18
   |
29 |         let vs = VarBuilder::zeros(DType::F32, device)?;
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `VarBuilderArgs<'_, Box<dyn SimpleBackend>>`
   |
   = help: the trait `Try` is not implemented for `VarBuilderArgs<'_, Box<dyn SimpleBackend>>`

error[E0061]: this function takes 2 arguments but 3 arguments were supplied
  --> src/model.rs:32:25
   |
32 |         let embedding = Linear::new(
   |                         ^^^^^^^^^^^
33 |             config.vocab_size,
   |             ----------------- expected `Tensor`, found `usize`
34 |             config.hidden_dim,
35 |             vs.pp("embedding"),
   |             ------------------ unexpected argument #3
   |
note: expected `Option<Tensor>`, found `usize`
  --> src/model.rs:34:13
   |
34 |             config.hidden_dim,
   |             ^^^^^^^^^^^^^^^^^
   = note: expected enum `std::option::Option<candle_core::Tensor>`
              found type `usize`
note: associated function defined here
  --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-nn-0.8.4/src/linear.rs:29:12
   |
29 |     pub fn new(weight: Tensor, bias: Option<Tensor>) -> Self {
   |            ^^^
help: remove the extra argument
   |
33 ~             /* candle_core::Tensor */,
34 ~             /* std::option::Option<candle_core::Tensor> */,
   |

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> src/model.rs:32:25
   |
32 |           let embedding = Linear::new(
   |  _________________________^
33 | |             config.vocab_size,
34 | |             config.hidden_dim,
35 | |             vs.pp("embedding"),
36 | |         )?;
   | |__________^ the `?` operator cannot be applied to type `candle_nn::Linear`
   |
   = help: the trait `Try` is not implemented for `candle_nn::Linear`

error[E0061]: this function takes 2 arguments but 3 arguments were supplied
  --> src/model.rs:41:25
   |
41 |             let layer = Linear::new(
   |                         ^^^^^^^^^^^
42 |                 config.hidden_dim,
   |                 ----------------- expected `Tensor`, found `usize`
43 |                 config.hidden_dim,
44 |                 vs.pp(format!("layer_{}", i)),
   |                 ----------------------------- unexpected argument #3
   |
note: expected `Option<Tensor>`, found `usize`
  --> src/model.rs:43:17
   |
43 |                 config.hidden_dim,
   |                 ^^^^^^^^^^^^^^^^^
   = note: expected enum `std::option::Option<candle_core::Tensor>`
              found type `usize`
note: associated function defined here
  --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-nn-0.8.4/src/linear.rs:29:12
   |
29 |     pub fn new(weight: Tensor, bias: Option<Tensor>) -> Self {
   |            ^^^
help: remove the extra argument
   |
42 ~                 /* candle_core::Tensor */,
43 ~                 /* std::option::Option<candle_core::Tensor> */,
   |

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> src/model.rs:41:25
   |
41 |               let layer = Linear::new(
   |  _________________________^
42 | |                 config.hidden_dim,
43 | |                 config.hidden_dim,
44 | |                 vs.pp(format!("layer_{}", i)),
45 | |             )?;
   | |______________^ the `?` operator cannot be applied to type `candle_nn::Linear`
   |
   = help: the trait `Try` is not implemented for `candle_nn::Linear`

error[E0061]: this function takes 2 arguments but 3 arguments were supplied
  --> src/model.rs:50:27
   |
50 |         let output_head = Linear::new(
   |                           ^^^^^^^^^^^
51 |             config.hidden_dim,
   |             ----------------- expected `Tensor`, found `usize`
52 |             config.vocab_size,
53 |             vs.pp("output"),
   |             --------------- unexpected argument #3
   |
note: expected `Option<Tensor>`, found `usize`
  --> src/model.rs:52:13
   |
52 |             config.vocab_size,
   |             ^^^^^^^^^^^^^^^^^
   = note: expected enum `std::option::Option<candle_core::Tensor>`
              found type `usize`
note: associated function defined here
  --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-nn-0.8.4/src/linear.rs:29:12
   |
29 |     pub fn new(weight: Tensor, bias: Option<Tensor>) -> Self {
   |            ^^^
help: remove the extra argument
   |
51 ~             /* candle_core::Tensor */,
52 ~             /* std::option::Option<candle_core::Tensor> */,
   |

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> src/model.rs:50:27
   |
50 |           let output_head = Linear::new(
   |  ___________________________^
51 | |             config.hidden_dim,
52 | |             config.vocab_size,
53 | |             vs.pp("output"),
54 | |         )?;
   | |__________^ the `?` operator cannot be applied to type `candle_nn::Linear`
   |
   = help: the trait `Try` is not implemented for `candle_nn::Linear`

error[E0599]: no method named `clamp_min` found for struct `candle_core::Tensor` in the current scope
    --> src/model.rs:82:29
     |
82   |             hidden = hidden.clamp_min(&Tensor::new(&[0.0_f32], &self.device)?)?;
     |                             ^^^^^^^^^
     |
help: there is a method `clamp` with a similar name, but with different arguments
    --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-core-0.8.4/src/tensor.rs:1108:5
     |
1108 |     pub fn clamp<T1: TensorOrScalar, T2: TensorOrScalar>(&self, min: T1, max: T2) -> Result<Self> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `unwrap` found for struct `VarBuilderArgs` in the current scope
   --> src/model.rs:114:62
    |
114 |         let vs = VarBuilder::zeros(DType::F32, &self.device).unwrap();
    |                                                              ^^^^^^ method not found in `VarBuilderArgs<'_, Box<dyn SimpleBackend>>`

error[E0308]: arguments to this function are incorrect
  --> src/trainer.rs:26:25
   |
26 |         let optimizer = AdamW::new(params, learning_rate)?;
   |                         ^^^^^^^^^^         ------------- expected `ParamsAdamW`, found `f64`
   |
note: expected `Vec<Var>`, found `Vec<Tensor>`
  --> src/trainer.rs:26:36
   |
26 |         let optimizer = AdamW::new(params, learning_rate)?;
   |                                    ^^^^^^
   = note: expected struct `Vec<Var>`
              found struct `Vec<candle_core::Tensor>`
note: associated function defined here
  --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-nn-0.8.4/src/optim.rs:8:8
   |
8  |     fn new(vars: Vec<Var>, config: Self::Config) -> Result<Self>;
   |        ^^^

error[E0277]: the trait bound `&Vec<i64>: NdArray` is not satisfied
   --> src/trainer.rs:155:13
    |
154 |         let input_ids = Tensor::new(
    |                         ----------- required by a bound introduced by this call
155 |             &flat.iter().map(|&x| x as i64).collect::<Vec<_>>(),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `WithDType` is not implemented for `&Vec<i64>`
    |
    = note: required for `&Vec<i64>` to implement `NdArray`
note: required by a bound in `candle_core::Tensor::new`
   --> /home/sysadmin/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/candle-core-0.8.4/src/tensor.rs:366:19
    |
366 |     pub fn new<A: crate::device::NdArray>(array: A, device: &Device) -> Result<Self> {
    |                   ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Tensor::new`
help: consider dereferencing here
    |
155 |             &*flat.iter().map(|&x| x as i64).collect::<Vec<_>>(),
    |              +

warning: use of deprecated method `rand::Rng::gen_range`: Renamed to `random_range`
  --> src/scheduler.rs:34:31
   |
34 |             let num_ops = rng.gen_range(2..std::cmp::min(5, num_machines + 1));
   |                               ^^^^^^^^^

warning: use of deprecated method `rand::Rng::gen_range`: Renamed to `random_range`
  --> src/scheduler.rs:39:37
   |
39 |                     machine_id: rng.gen_range(0..num_machines),
   |                                     ^^^^^^^^^

warning: use of deprecated method `rand::Rng::gen_range`: Renamed to `random_range`
  --> src/scheduler.rs:40:35
   |
40 |                     duration: rng.gen_range(1..20),
   |                                   ^^^^^^^^^

warning: unused import: `rand::Rng`
 --> src/main.rs:8:5
  |
8 | use rand::Rng;
  |     ^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused variable: `seed`
  --> src/scheduler.rs:29:57
   |
29 |     pub fn random(num_jobs: usize, num_machines: usize, seed: u64) -> Self {
   |                                                         ^^^^ help: if this is intentional, prefix it with an underscore: `_seed`
   |
   = note: `#[warn(unused_variables)]` on by default

Some errors have detailed explanations: E0061, E0277, E0308, E0599.
For more information about an error, try `rustc --explain E0061`.
warning: `jssp-scheduler-real` (bin "jssp-scheduler") generated 7 warnings
error: could not compile `jssp-scheduler-real` (bin "jssp-scheduler") due to 11 previous errors; 7 warnings emitted
